zipkin-server/src/main/java/zipkin/server/ZipkinServer.java

```diff
@@ -13,12 +13,27 @@
  */
 package zipkin.server;
 
-import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.SpringBootConfiguration;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
 import org.springframework.boot.builder.SpringApplicationBuilder;
 import zipkin2.server.internal.EnableZipkinServer;
 import zipkin2.server.internal.banner.ZipkinBanner;
 
-@SpringBootApplication
+/**
+ * This adds the {@link EnableAutoConfiguration} annotation, but disables it by default to save
+ * startup time. A supported integration will need to explicitly enable auto-configuration like so,
```

Comment:
This seems fragile to me, and might break unexpectedly if the main app comes to rely on the fact that it has no autoconfiguration. Maybe there is a better way using a custom autoconfiguration annotation (so that supported integrations only need to add that annotation).

Reply:
the implementation has changed. our modules no longer rely on autoconfiguration at all. most of the time I wouldn't expect anyone to enable auto-configuration.